src/hotspot/share/opto/mulnode.cpp

```diff
@@ -1401,9 +1448,9 @@ Node *RShiftINode::Ideal(PhaseGVN *phase, bool can_reshape) {
   }
 
   // Check for "(byte[i] <<24)>>24" which simply sign-extends
-  if( shift == 24 &&
-      (t3 = phase->type(shl->in(2))->isa_int()) &&
-      t3->is_con(24) ) {
+  if (shift == 24 &&
+      (left_shift_t = phase->type(shl->in(2))->isa_int()) &&
```

Comment:
Then you could also refactor down here.
Because these are implicit nullptr checks again.